Userland/Applications/NetworkSettings/NetworkSettingsWidget.cpp

```diff
@@ -134,27 +138,64 @@ void NetworkSettingsWidget::on_switch_enabled_or_dhcp()
 
 void NetworkSettingsWidget::apply_settings()
 {
-    auto config_file = Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes).release_value_but_fixme_should_propagate_errors();
+    int wpipefds[2];
+    if (pipe2(wpipefds, O_CLOEXEC) < 0) {
```

Comment:
Although `apply_settings()` does not yet return an `ErrorOr<T>` using `Core::System::pipe2()` would probably be better and either do a `release_value_but_fixme_should_propagate_errors()` or something like this:
```cpp
if (result.is_error()) {
    GUI::MessageBox::show_error(window(), String::formatted("{}", result.release_error()).release_value_but_fixme_should_propagate_errors());
    return;
}
```